src/preferences/dialog/dlgprefwaveform.cpp

```diff
@@ -24,11 +24,15 @@ DlgPrefWaveform::DlgPrefWaveform(QWidget* pParent, MixxxMainWindow* pMixxx,
 
     // Populate waveform options.
     WaveformWidgetFactory* factory = WaveformWidgetFactory::instance();
+    // We assume that the 'handles' origin list remains constant.
+    // We will use the handles index later on to set waveform types and to
+    // update the combobox.
     QVector<WaveformWidgetAbstractHandle> handles = factory->getAvailableTypes();
     for (int i = 0; i < handles.size(); ++i) {
-        waveformTypeComboBox->addItem(handles[i].getDisplayName(),
-                                      handles[i].getType());
+        waveformTypeComboBox->addItem(handles[i].getDisplayName(), i);
```

Comment:
Use the integral number of the enum here, QVector and index are not needed.

Then extract it with QVariant() -> int -> enum later.

Reply:
> Then extract it with QVariant() -> int -> enum later.

yes, that would be nicer. I tried that, and I could also watch the enum being passed to and inside the factory correctly as before. the new type was saved to config BUT not applied immediatley as before, only after reading again from config, so I guess it was some type mismatch down the line but I did not get any compile errors or warnings about that.
the reason I chose the current solution is that all the index -> type and type -> index conversion can happen in the factory where the list is stored.

Reply:
> QVector and index are not needed

indeed. what you mean is
```
    foreach (const auto handle, factory->getAvailableTypes()) {
        waveformTypeComboBox->addItem(handle.getDisplayName(), static_cast<int>(handle.getType()));
    }
```
or
```
    foreach (const auto handle, factory->getAvailableTypes()) {
        waveformTypeComboBox->addItem(handle.getDisplayName(), handle.getType());
    }
```
then we _could_ use [WaveformWidgetFactory::setWidgetType(WaveformWidgetType::Type type)](https://github.com/mixxxdj/mixxx/blob/f4da747e974e1ed2ee1c82c4c09b8777265ff6d3/src/waveform/waveformwidgetfactory.cpp#L450).

BUT
> the new type was saved to config BUT not applied immediatley as before

...because only [factory->setWidgetTypeFromHandle](https://github.com/mixxxdj/mixxx/blob/f4da747e974e1ed2ee1c82c4c09b8777265ff6d3/src/waveform/waveformwidgetfactory.cpp#L493) would actually update the widgets immediately.


Given that, I'd stick with consistently using the handle index in the preferences and the type in the factory,  to avoid refactoring the factory:
```
    QVector<WaveformWidgetAbstractHandle> handles = factory->getAvailableTypes();
    for (int i = 0; i < handles.size(); ++i) {
        waveformTypeComboBox->addItem(handles[i].getDisplayName(), i);
    }
```
and `factory->setWidgetTypeFromHandle(int)`.

Reply:
Choose whatever is more comfortable for you.